src/abstractdataframe/selection.jl

```diff
@@ -10,6 +10,148 @@
+    rows. As a particular rule, values wrapped in a `Ref` or a `0`-dimensional
+    `AbstractArray` are unwrapped and then treated as a single row.
+
+    `select`/`select!` and `transform`/`transform!` always return a `DataFrame`
+    with the same number and order of rows as the source (even if `GroupedDataFrame`
+    had its groups reordered).
+
+    For `combine`, rows in the returned object appear in the order of groups in
+    the `GroupedDataFrame`. The functions can return an arbitrary number of rows
+    for each group, but the kind of returned object and the number and names of
+    columns must be the same for all groups, except when a `DataFrame()` or
+    `NamedTuple()` is returned, in which case a given group is skipped.
+
+    It is allowed to mix single values and vectors if multiple transformations
+    are requested. In this case single value will be repeated to match the length
+    of columns specified by returned vectors.
+
+    To apply `function` to each row instead of whole columns, it can be wrapped in a
+    `ByRow` struct. `cols` can be any column indexing syntax, in which case
+    `function` will be passed one argument for each of the columns specified by
+    `cols` or a `NamedTuple` of them if specified columns are wrapped in `AsTable`.
+    If `ByRow` is used it is allowed for `cols` to select an empty set of columns,
+    in which case `function` is called for each row without any arguments and an
+    empty `NamedTuple` is passed if empty set of columns is wrapped in `AsTable`.
+
+    If a collection of column names is passed then requesting duplicate column
+    names in target data frame are accepted (e.g. `select!(df, [:a], :, r"a")`
+    is allowed) and only the first occurrence is used. In particular a syntax to
+    move column `:col` to the first position in the data frame is
+    `select!(df, :col, :)`. On the contrary, output column names of renaming,
+    transformation and single column selection operations must be unique, so e.g.
+    `select!(df, :a, :a => :a)` or `select!(df, :a, :a => ByRow(sin) => :a)` are not allowed.
+
+    Note that including the same column several times in the data frame via renaming
+    or transformations that return the same object without copying may create
+    column aliases even if `copycols=true`. An example of such a situation is
+    `select!(df, :a, :a => :b, :a => identity => :c)`.
+
+    If `df` is a `SubDataFrame` and `copycols=true` then a `DataFrame` is
+    returned and the same copying rules apply as for a `DataFrame` input: this
+    means in particular that selected columns will be copied. If
+    `copycols=false`, a `SubDataFrame` is returned without copying columns.
+
+    There the following keyword arguments are supported by the transformation functions
+    (not all keyword arguments are supported in all cases; in general they are allowed
+    in situations when they are meaningful, see the documentation of the specific functions
+    for details):
```

Comment:
changed to "signatures" as I show it there. I would leave it in one place. It will be a nightmare to maintain it in several places (you have already requested above to merge the documentation between docstrings and manual :smile:).

Reply:
But how about making this part specific to each function's docstring so that we never mention an argument that a function doesn't support?

(BTW I don't see the change to "signature".)

Reply:
OK - I moved them to specific functions.